All_home_works/hw3/task3.py

```diff
@@ -0,0 +1,48 @@
+# I decided to write a code that generates data filtering object from a list of keyword parameters:
+
+
+class Filter:
+    """
+    Helper filter class. Accepts a list of single-argument
+    functions that return True if object in list conforms to some criteria
+    """
+
+    def __init__(self, *functions):
+        self.functions = functions
+
+    def apply(self, data):
+        return [item for item in data if all(i(item) for i in self.functions)]
+
+
+# example of usage:
+# positive_even = Filter(lambda a: a % 1 == 0, lambda a: a > 0, lambda a: isinstance(int, a)))
+# positive_even.apply(range(99)) should return only even numbers from 0 to 99
+
+
+def make_filter(**kwargs):
+    """
+    Generate filter object for specified keywords
+    """
+    filter_funcs = []
+    for key, value in kwargs.items():
+
+        def keyword_filter_func(checking_dict):
+            return checking_dict[key] == value if key in checking_dict else False
+
+        filter_funcs.append(keyword_filter_func)
+    return Filter(*filter_funcs)
+
+
+sample_data = [
```

Comment:
This is example data, no need to leave it here with function. Just move it to tests.

Reply:
Fixed

Reply:
Have you moved it?

Reply:
I copied not moved it into the test because it was in a description and I did not want to touch it...
Now this data only in the test.

Reply:
> I copied not moved it into the test because it was in a description and I did not want to touch it...

If I ask you something and you disagree, it is ok to say that. When you respond "fixed" and do something else, it looks very strange.